### ISSUE TITLE
[openh264] static build (VCPKG_LIBRARY_LINKAGE static) still install libopenh264.so

### ISSUE THREAD

**Issue by stalkyr** [microsoft/vcpkg#17964]
**Host Environment**
- OS: Ubuntu
- Triplet: x64-linux

**To Reproduce**
`./vcpkg install openh264`
check installed/x64-linux/lib folder

**Additional context**
Because vcpkg install libopenh264.so, cmake will pick that file to `openh264_LIBRARY` first, make user executable link dynamically instead of statically

Manually delete .so files and re-generate cmake  can solve this problem

**Comment by JackBoosY** [comment]
Hi @stalkyr, thanks for reporting this issue!
But I found there is nothing wrong with the linux build:
```bash
root@vzayUb:/home/work/vcpkg# ll packages/openh264_x64-linux/lib/
total 1972
drwxr-xr-x 3 root root    4096 May 19 06:31 ./
drwxr-xr-x 6 root root    4096 May 19 06:31 ../
-rw-r--r-- 1 root root 2003830 May 19 06:31 libopenh264.a
drwxr-xr-x 2 root root    4096 May 19 06:31 pkgconfig/
```

**Comment by mcmtroffaes** [comment]
This should have been recently fixed with #17592.

**Comment by JackBoosY** [comment]
Please check your vcpkg whether it's latest.
If you still have trouble here, please reopen this issue.

Thanks.
